src/android.ts

```diff
@@ -162,11 +144,31 @@ export const androidApi = <RunTarget extends SupportedRunTarget<'android'>>(
             );
             if (!fridaIsStarted) throw new Error('Failed to start Frida.');
         },
+        async hasDeviceBooted(waitForDevice = false) {
+            const { stdout: devBootcomplete } = await adb(
+                [...(waitForDevice ? ['wait-for-device'] : []), 'shell', 'getprop', 'dev.bootcomplete'],
+                {
+                    reject: false,
+                    timeout: 200,
+                }
+            );
+            const { stdout: sysBootCompleted } = await adb(
+                [...(waitForDevice ? ['wait-for-device'] : []), 'shell', 'getprop', 'sys.boot_completed'],
+                { reject: false, timeout: 200 }
+            );
+            const { stdout: bootanim } = await adb(
+                [...(waitForDevice ? ['wait-for-device'] : []), 'shell', 'getprop', 'init.svc.bootanim'],
+                {
+                    reject: false,
+                    timeout: 200,
+                }
+            );
+            return devBootcomplete.includes('1') && sysBootCompleted.includes('1') && bootanim.includes('stopped');
+        },
         async requireRoot(action) {
             if (!options.capabilities.includes('root')) throw new Error(`Root access is required for ${action}.`);
 
             await adb(['root']);
-            await this.awaitAdb();
```

Comment:
Why? I didn‘t understand this. Do we need to wait for the device to be booted here or just for adb to work? Generally this runs after `ensureDevice`, doesn‘t it?

Reply:
Waiting for `adb get-state` as we did [previously](https://github.com/tweaselORG/appstraction/commit/92cdda7c165dca1b5316ee4600602077eb3c4b46#diff-34632e78f038aee01b56eebd9e73483544c8127003466eb473ca92f91cad74f1R36-R40) would be enough. Checking the boot properties will also work since these implicitly wait for adb to be ready, but it's not the right thing semantically.

Reply:
Oh wait, I guess just `adb wait-for-device` should also be enough.